builder/graph/graph.go

```diff
@@ -0,0 +1,126 @@
+package graph
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"text/template"
+)
+
+// Node defines a single unit containing build information about a file.
+type Node struct {
+	// Unique node name
+	Name string
+	// Nodes will describe nodes connected to this one
+	Nodes map[string]*Node
+	// Builder specific fields
+	Template template.Template
+	Links    map[string]interface{}
+}
+
+// NewNode create a empty Node.
+func NewNode(name string) *Node {
+	return &Node{
+		Name:  name,
+		Nodes: map[string]*Node{},
+		Links: map[string]interface{}{},
+	}
+}
+
+// Graph defines a collection of Nodes.
+type Graph struct {
```

Comment:
Maybe worth adding to the code comment how this Graph struct is used in the sense of does it represent a single OCI Artifact, or is it used to load multiple artifacts? The graph defines connections between keys, but it does not contain any other information per key, such as it's location, content, size, etc which is defined in ... (?)